### DISCUSSION TITLE
Linux: using `brew shellenv` causes `sudo` to stop working.

### DISCUSSION THREAD

**Discussion by cyrneko** [Homebrew/discussions#4414] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.13
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9b60fcf7ac3f82fa589a64091376d0663262f8c6
Last commit: 6 hours ago
Core tap JSON: 12 Apr 22:06 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 3
Homebrew Ruby: 2.6.10 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: 3-core 64-bit amd_k10
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.0-69-generic x86_64 GNU/Linux
OS: Linux Mint 21.1 (vera)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I've noticed that after adding `eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"` to my `.bashrc` or my `.zshrc`, that `sudo` stops working entirely, telling me that "_`The "no new privileges" flag is set, which prevents sudo from running as root.`_"

I haven't been able to 100% pin down what causes this, but the line has been in [my `.zshrc`](https://codeberg.org/TGRush/dotfiles/src/branch/catppuccin/.zshrc) for months now and never caused trouble before.

Has the line that needs to be added changed? or is this some weird side-effect of a recent change? Any help would be appreciated greatly! :smile:

**Comment by ZhongRuoyu** [comment]
This doesn't sound to be caused by `brew shellenv` -- does `sudo` work for you if you comment out the `eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"` line? (For example, does `sudo echo hello` print out `hello`?)

Are you running this in a Docker? If so, did you start the container with any [security options](https://docs.docker.com/engine/reference/run/#security-configuration), like `--security-opt="no-new-privileges=true"`? From [`sudo`'s manual](https://www.sudo.ws/docs/man/sudo.man/#The):

> `The "no new privileges" flag is set, which prevents sudo from running as root.`
>
> `sudo` was run by a process that has the Linux “no new privileges” flag is set. This causes the set-user-ID bit to be ignored when running an executable, which will prevent `sudo` from functioning. The most likely cause for this is running `sudo` within a container that sets this flag. Check the documentation to see if it is possible to configure the container such that the flag is not set.

**Reply by cyrneko** [reply]
Nope, all bare-metal, and it does go away when I comment out the line from my .zshrc/.bashrc file

Altough, after prolonged use I have noticed that the behaviour with this issue is incredibly inconsistent. Sometimes it works fine, sometimes `sudo` stops working, yet a consistent fix continues to just be commenting out the line 🤔 

I will look further into this, and see if there‘s any specific Terminal Application or else which conflicts.

**Comment by gromgit** [comment]
What's the output of the following *with `brew shellenv` active*?
```
echo $PATH
which -a sudo
```
